R-package/R/xgb.Booster.R

```diff
@@ -641,31 +540,22 @@ xgb.attributes <- function(object) {
   if (is.null(names(a)) || any(nchar(names(a)) == 0)) {
     stop("attribute names cannot be empty strings")
   }
-  # Coerce the elements to be scalar strings.
-  # Q: should we warn a user about non-scalar elements?
-  a <- lapply(a, function(x) {
-    if (is.null(x)) return(NULL)
-    if (is.numeric(x[1])) {
-      format(x[1], digits = 17)
-    } else {
-      as.character(x[1])
-    }
-  })
-  handle <- xgb.get.handle(object)
   for (i in seq_along(a)) {
-    .Call(XGBoosterSetAttr_R, handle, names(a[i]), a[[i]])
-  }
-  if (is(object, 'xgb.Booster') && !is.null(object$raw)) {
-    object$raw <- xgb.serialize(object$handle)
+    xgb.attr(object, names(a[i])) <- a[[i]]
   }
-  object
+  return(object)
 }
 
-#' Accessors for model parameters as JSON string
+#' @title Accessors for model parameters as JSON string
+#' @details Note that assignment is performed in-place on the booster C object, which unlike assignment
+#' of R attributes, doesn't follow typical copy-on-write semantics for assignment - i.e. all references
+#' to the same booster will also get updated.
 #'
-#' @param object Object of class `xgb.Booster`.
+#' See \link{xgb.copy.Booster} for an example of this behavior.
+#' @param object Object of class `xgb.Booster`. \bold{Will be modified in-place} when assigning to it.
 #' @param value A JSON string.
-#'
+#' @return `xgb.config` will return the parameters as JSON text, which can be parsed through
+#' e.g. `jsonlite::fromJSON`.
```

Comment:
I regret that I have made the interface like this, I think it's much better if we can call jsonlite inside this function. What do you think?

Reply:
Switched to R lists instead of JSON text.

Reply:
Thank you for the change!